### PR TITLE
fight timebound quota: try to schedule a pod multiple times

### PR DESCRIPTION
**Pull request by TomasTomecek**
with exponential backoff to try to fit the timebound quota

The timebound quota is really hard to predict (because it accounts
openshift objects outside of our control such as deploy pods) and
constantly jumps up and down - failing to schedule a pod because we
didn't fit in the right time is "too expensive" when we can try the next
time and probably get scheduled.

we'll try this often:
```
In [1]: 3**1
Out[1]: 3

In [2]: 3**2
Out[2]: 9

In [3]: 3**3
Out[3]: 27

In [4]: 3**4
Out[4]: 81

In [5]: 3**5
Out[5]: 243
```

### CODE REVIEW

File: sandcastle/api.py [hunk 1]
```diff
@@ -321,9 +321,26 @@ def create_pod(self, pod_manifest: Dict) -> Dict:
 
         :return: response from the API server
         """
-        return self.api.create_namespaced_pod(
-            body=pod_manifest, namespace=self.k8s_namespace_name
-        )
+        idx = 1
+        # if we hit timebound quota, let's try 5 times with expo backoff
+        while idx < 6:
```
Comment (csomh) [inline sandcastle/api.py:326]: Wouldn't it be clearer to use a for-loop here?

**Review by lachmanfrantisek — APPROVED** [review]
Nice!

I am only thinking if some already existing `retry` decorator can't do that for us. But I see that we need to check for the specific status number so that might not be possible to do.

File: sandcastle/api.py (hunk 1 above)
Comment (lachmanfrantisek) [reply sandcastle/api.py:326]: And `for-else`?

File: sandcastle/api.py [hunk 2]
```diff
@@ -321,9 +321,26 @@ def create_pod(self, pod_manifest: Dict) -> Dict:
 
         :return: response from the API server
         """
-        return self.api.create_namespaced_pod(
-            body=pod_manifest, namespace=self.k8s_namespace_name
-        )
+        idx = 1
+        # if we hit timebound quota, let's try 5 times with expo backoff
```
Comment (lachmanfrantisek) [inline sandcastle/api.py:325]: And btw, what about using some constant?

**Pull request merged**